components/api_version/src/api_v1.rs

```diff
@@ -35,4 +35,35 @@ impl KvFormat for ApiV1 {
     fn encode_raw_value_owned(value: RawValue<Vec<u8>>) -> Vec<u8> {
         value.user_value
     }
+
+    fn convert_encoded_key_version_from(
+        src_api: ApiVersion,
+        key: &[u8],
+        _ts: Option<TimeStamp>,
+    ) -> Result<Key> {
+        match src_api {
+            ApiVersion::V1 | ApiVersion::V1ttl => Ok(Key::from_encoded_slice(key)),
+            ApiVersion::V2 => {
+                assert_eq!(ApiV2::parse_key_mode(key), KeyMode::Raw);
```

Comment:
Should we return an error instead of assert?

Reply:
I would like to change the assert to debug_assert. As it's the rare case to happend and it's a little performance consuming. How do you think?